storops/unity/resource/lun.py

```diff
@@ -257,8 +263,35 @@ def delete(self, async_mode=False, force_snap_delete=False,
                                 forceSnapDeletion=force_snap_delete,
                                 forceVvolDeletion=force_vvol_delete,
                                 async_mode=async_mode)
+
+        @retryz.retry(timeout=async_timeout, wait=async_interval,
+                      on_return=False)
+        def _do_update(jh, job_id):
+            job = jh.get_job(job_id)
+
+            if job is None:
+                return False
+
+            if job.state == enums.JobStateEnum.COMPLETED:
+                return job
+            elif job.state in (enums.JobStateEnum.FAILED,
+                               enums.JobStateEnum.ROLLING_BACK,
+                               enums.JobStateEnum.COMPLETED_WITH_ERROR):
+                raise ex.UnityDeleteLunJobError(job)
+            return False
+
         try:
             resp.raise_if_err()
+
+            if async_mode and resp.job.existed:
+                jh = job_helper.get_job_helper(self._cli)
+                job_id = resp.job.id
+                jh.add_job(job_id)
+                job = _do_update(jh, job_id)
+                jh.remove_job(job_id)
+                # TODO: need to figure out how to update resp.job in a good way.
```

Comment:
Did you try `resp.job.update()`?

Reply:
Keep using resp.job = xxx, reasons:
1. resp.job.update() doesn't work now, need to fix it
2. resp.job.update() will trigger a new GET request which may impact system performance in large concurrency